rig-core/src/completion/message.rs

```diff
@@ -71,6 +71,8 @@ pub enum ToolResultContent {
 #[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
 pub struct ToolCall {
     pub id: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub index: Option<usize>,
```

Comment:
I'm not sure I understand what this index is referencing, is this an important field? It seems to me if there's an id to the tool call, there's no need for an index if there are multiple tool calls.

I think it shouldn't be in this PR as it would require updates to other providers like anthropic, etc.

Reply:
JSON has ordering in lists, so i'm not entirely sure if it's necessary. Maybe i'll check up with the docs, sometimes these providers do really odd things with their modelling.

Reply:
I removed everything that is not related with DeepSeek, if we get into issues with the index in future it will be an easy PR